### DISCUSSION TITLE
Type hinting with uvloop and Windows

### DISCUSSION THREAD

**Discussion by pmdevita** [python/typing#1466] [General]
Hey, I have a project that conditionally installs and loads uvloop when it is running on Linux and has uvloop installed. The trouble is, you can't install uvloop on Windows, so type checking fails. To fix this, you can add a `# type: ignore` to the line but then it fails on Linux with `error: Unused "type: ignore" comment  [unused-ignore]`.

Here's the [snippet in question](
https://github.com/pmdevita/hikari-atsume/blob/932d57e6f966e44faa752c186810a1f0583bf029/atsume/bot.py#L47-L56):

```python
        try:
            # Has to be ignored to keep tests passing on Windows
            import uvloop  # type: ignore


            uvloop.install()
            logging.info("Enabling uvloop...")
        except ImportError:
            pass
```

How should I handle this? Thanks for the help!

**Comment by hauntsaninja** [comment]
One solution is to gate with `if sys.platform != "win32"`.

Another solution that works with recent mypy is you can do something like `import uvloop  # type: ignore[import, unused-ignore]`. See also https://github.com/python/mypy/issues/8823#issuecomment-1556288015

**Comment by pmdevita** [comment]
Ah I didn't know mypy will inference if statements like that. I added that if statement and it works, thank you so much!
